src/unix/atomic-ops.h

```diff
@@ -49,7 +49,7 @@ UV_UNUSED(static int cmpxchgi(int* ptr, int oldval, int newval)) {
   else
     return op4;
 #elif defined(__SUNPRO_C) || defined(__SUNPRO_CC)
-  return atomic_cas_uint(ptr, oldval, newval);
+  return atomic_cas_uint((uint_t *)ptr, (uint_t)oldval, (uint_t)newval);
```

Comment:
I suppose that, rather than casting pointers, this is better fixed by changing the function prototype to operate on `unsigned int` and update the call sites and types (`uv_async_t`, `uv_spinlock_t`) accordingly.

Likewise for `cmpxchgl()`. We don't actually use that anywhere right now so I'd also be okay with just removing it.

Reply:
Are you okay with accepting the single line cast patch as part of this series (`__MVS__` already performs the same casts) and then following up a subsequent PR to address what you said? My main concern with changing the signature as part of this series is that it touches all the other architectures and I don't have a good way to test those branches. It's also a pretty beefy patch on its own.

Reply:
Yes, that's fine.